MyOwnORM/MyOwnORM/Implementations/CustomDbSetRepository.cs

```diff
@@ -0,0 +1,536 @@
+                string idValue = string.Empty;
+
+                using (SqlDataReader reader = await command.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        T entity = dbSetReflection.CreateInstanceType();
+
+                        dbSetReflection.SetPropertiesForIncludeExpressionsMethod(reader, idValue, entity);
+
+                        foreach (var include in includes)
+                        {
+                            List<string> propVal = dbSetReflection.GetPropertyValues(new Expression<Func<T, object>>[] { include });
+                            object[] includeType = dbSetReflection.GetIncludeTypes(new Expression<Func<T, object>>[] { include });
+
+                            for (int i = 0; i < propVal.Count; i++)
+                            {
+                                PropertyInfo propertyInfo = dbSetReflection.GetPropertyInfo(propVal[i]);
+                                Type propertyType = propertyInfo.PropertyType;
+
+                                if (dbSetReflection.IsCollectionType(propertyInfo))
+                                {
+                                    SetGenericEntityIncludeExpressionsMethod(propertyInfo, includeType[i].GetType(), entity, idValue, reader);
+                                }
+                                else
+                                {
+                                    SetEntityIncludeExpressionsMethod(includeType[i].GetType(), entity, propertyInfo, idValue, reader, propertyType);
+                                }
+                            }
+                        }
+
+                        entities.Add(entity);
+                    }
+                }
+            }
+            return entities;
+        }
+
+        public async Task<IQueryable<T>> WhereAsync(Expression<Func<T, bool>> predicate)
+        {
+            List<T> entities = new List<T>();
+            using (SqlConnection connection = new SqlConnection(_connectionString))
+            {
+                await connection.OpenAsync();
+
+                string key = dbSetExtension.GetKeyInLambdaExpression(predicate);
+                object val = dbSetExtension.GetValueInLambdaExpression(predicate);
+
+                query = $"SELECT * FROM {tableName} WHERE {key} = {val}";
+                using (SqlCommand command = new SqlCommand(query, connection))
+                {
+                    using (SqlDataReader reader = await command.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            T entity = dbSetReflection.MapReaderToEntity(reader);
+                            entities.Add(entity);
+                        }
+                    }
+                }
+            }
+
+            return entities.AsQueryable();
+        }
+
+        public async Task<T> GetByIdAsync<TKey>(TKey id)
```

Comment:
you must apply generic to whole class, because not method specifies type of key. It is repository purpose. Change everywhere

Reply:
Chaged